### PR TITLE
Add simplification of long if based on tip from @jonmv.

### PR DESCRIPTION
**Pull request by baldersheim**
@jonmv PR

### CODE REVIEW

File: vespajlib/src/main/java/com/yahoo/text/Text.java [hunk 1]
```diff
@@ -56,38 +56,8 @@ private static boolean isTextCharAboveMinSurrogate(int codepoint) {
...
-        if (codepoint <= 0xCFFFF)  return false;
-        if (codepoint <  0xDFFFE)  return true;
-        if (codepoint <= 0xDFFFF)  return false;
-        if (codepoint <  0xEFFFE)  return true;
-        if (codepoint <= 0xEFFFF)  return false;
-        if (codepoint <  0xFFFFE)  return true;
-        if (codepoint <= 0xFFFFF)  return false;
-        if (codepoint <  0x10FFFE) return true;
-        return false;
+        if (codepoint >= 0x10FFFE) return false;
```
Comment (jonmv) [inline vespajlib/src/main/java/com/yahoo/text/Text.java:59]:
```suggestion
        if (codepoint < 0x10000) return false;
        if (codepoint >= 0x10FFFE) return false;
```

Comment (jonmv) [reply vespajlib/src/main/java/com/yahoo/text/Text.java:59]: Like so

Comment (baldersheim) [reply vespajlib/src/main/java/com/yahoo/text/Text.java:60]: Correct, but it should return true.... I guess

Comment (jonmv) [reply vespajlib/src/main/java/com/yahoo/text/Text.java:60]: Haha, yes.

**Review by jonmv — APPROVED** [review]

**Comment by jonmv** [comment]
Do you have any idea why we're considering all these FFFE and FFFF values invalid, though?

**Comment by baldersheim** [comment]
I have no idea. I assume someone has read some formal specs. There are some links in the code.

**Pull request merged**